### ISSUE TITLE
How long & Single Object

### ISSUE THREAD

**Issue by gattia** [princeton-computational-imaging/Diffusion-SDF#5]
I am trying to use diffusion SDF for modeling a single object (where each instance is relatively similar to the other instances). I think the model is experiencing some sort of mode collapse and mostly producing an average shape (with some small variations of very easy-to-decipher features). Have you had any experience with this issue? Do you have any suggestions?

Related, what is the timeline for training? The readme says a few days to get to a certain loss. Is a few days usually required? Im currently doing tests with ~1 day (up to 10k epochs - only currently using around 2-300 meshes, which has worked successfully for DeepSDF before), am I maybe not waiting long enough? 

I've also tried playing with: 
- LR
- batch size
- KLD weight
- briefly/currently trying higher: 
    - plane resolution
    - point cloud size

Upto 1 day of training these all make small differences, but they mostly are converging on the same/similar solution.

**Comment by gattia** [comment]
Also, I get weird artefacts in all of my reconstructions - either if I used the zero vector, or I try to encode a mesh. 

The artefacts are such that I get a smooth surface in the quadrant [-1, -1, -1] and all of the other quadrants have jagged edges. Not sure what could be causing this - maybe something about re-ordering the planes, or sampling? 

In the below image, the bottom left is the -1, -1, -1 quadrant. It looks like the other quadrants have somewhat distinct slicing issues. 

![participant_femur_recon](https://user-images.githubusercontent.com/10676083/223290143-f202a971-2733-431b-91a3-65eafb634373.png)

I have re-created the code for diffusionsdf in my own project, but I have checked everything and think that it should all be the same.

**Comment by gene-chou** [comment]
For the first question, can I take a look at samples of the training data, stage 1 reconstructions, and stage 2/3 generations? Our VAE has substantially more parameters than DeepSDF so a longer training time would be expected. For the second question, I don't think I understand. Is the image a generated sample or a reconstruction from stage 1? By artifacts, are you referring to the unsmooth surfaces?

**Comment by gattia** [comment]
Thanks for the quick reply. 

I am only trying to train stage 1 right now. I have generated samples by either: 1) reconstruction from a point cloud, or 2) provide a latent vector to the VAE bottleneck and reconstruct that (I often use a vector of zeros to see what the "average" shape is). The reconstructed and average shapes look similar - they have the same smooth and unsmooth regions with the same patterns (even on different training runs with different training parameters). The attached image was a reconstruction of a mesh from an inputted point cloud. 

Yes, by artefacts I mean the unsmooth surfaces - it seems peculiar that it is smooth in one quadrant (<0, <0, <0) and unsmooth in specific ways in the other quadrants. This "artefact" seems to show up in all of my training runs and generations. 

And just to say this - that object does look "similar" to what I expect the objects to look like. It is a femur bone. It just doesn't seem to be doing great at reconstruction - e.g, the unsmooth areas, and it is very close to the average shape regardless of the example.

**Comment by gene-chou** [comment]
We didn't encounter this issue when training on single categories of the Acronym / ShapeNet datasets. Maybe your data is just too similar. We'd recommend starting with training on just 1 sample and see if the model is able to overfit such that there are no artifacts. If this works, then maybe some tuning will eventually make this work. Feel free to email me if you want to discuss further.
